llvm/lib/Target/RISCV/RISCVFeatures.td

```diff
@@ -10,119 +10,155 @@
 // RISC-V subtarget features and instruction predicates.
 //===----------------------------------------------------------------------===//
 
+// Subclass of SubtargetFeature to be used when the feature is also a RISC-V
+// extension. Extensions have a version and may be experimental.
+//
+// name      - Name of the extension in lower case.q
+// major     - Major version of extension
+// minor     - Minor version of extension
+// fieldname - name of field to create in RISCVSubtarget.
```

Comment:
This is can be standarized, we force the name of `RISCVExtension` to be in form of `FeatureXXX` and then the fieldname can be `!subst(NAME, "Feature", "Has")`.

Reply:
This doesn't work for Zvl*b.

Reply:
Thanks! Then I still think we can make it default to reduce some code:
```python
class RISCVExtension<string name, int major, int minor, string desc,
                     list<SubtargetFeature> implies = [],
                     string fieldname = !subst(NAME, "Feature", "Has"),
                     string value = "true">
```